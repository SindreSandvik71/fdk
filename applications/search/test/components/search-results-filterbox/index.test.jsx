import React from 'react';
import { shallow } from 'enzyme';
import FilterBox from '../../../src/components/search-results-filterbox';

let onClick;
let defaultProps;
let wrapper;

beforeEach(() => {
  onClick = jest.fn();
  defaultProps = {
    title: 'Testbox',
    filter: {
      buckets: [
        { key: 'GOVE', doc_count: 104 },
        { key: 'EDUC', doc_count: 11 },
        { key: 'HEAL', doc_count: 8 },
        { key: 'Ukjent', doc_count: 7 },
        { key: 'AGRI', doc_count: 1 }
      ]
    },
    onClick,
    activeFilter: null,
    themesItems: {
      AGRI: {
        id: 'http://publications.europa.eu/resource/authority/data-theme/AGRI',
        code: 'AGRI',
        startUse: '2015-10-01',
        title: {
          pt: 'Agricultura, pesca, silvicultura e alimentação',
          bg: 'Селско стопанство, рибарство, горско стопанство и храни'
        },
        conceptSchema: {
          id: 'http://publications.europa.eu/resource/authority/data-theme',
          title: { en: 'Dataset types Named Authority List' }
        }
      }
    }
  };
  wrapper = shallow(<FilterBox {...defaultProps} />);
});

test('should render FilterBox correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render FilterBox correctly with activeFilter', () => {
  wrapper.setProps({
    activeFilter: 'GOVE'
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render FilterBox correctly with activeFilter', () => {
  wrapper.setProps({
    filter: {}
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle toggleFilter', () => {
  wrapper.find('button.fdk-toggleFilter').simulate('click');
  expect(wrapper.state('openFilter')).toBe(false);
});

test('should handle toggleList', () => {
  wrapper.find('button.fdk-toggleList').simulate('click');
  expect(wrapper.state('open')).toBe(true);
});
