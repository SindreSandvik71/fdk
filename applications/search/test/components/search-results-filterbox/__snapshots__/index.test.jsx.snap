// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterBox correctly 1`] = `
<div
  className="fdk-panel--filter"
>
  <div
    className="fdk-panel__header"
  >
    <button
      className="fdk-toggleFilter fdk-p-0"
      onClick={[Function]}
    >
      <i
        className="fa mr-2 fa-angle-up"
      />
      <span>
        Testbox
      </span>
    </button>
  </div>
  <Collapse
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    transitionAppear={false}
    unmountOnExit={false}
  >
    <div
      className="fdk-panel__content"
    >
      <div
        className="fdk-items-list"
      >
        <div>
          <FilterOption
            active={false}
            count={104}
            displayClass={null}
            itemKey={0}
            key="0"
            label="GOVE"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="GOVE"
          />
          <FilterOption
            active={false}
            count={11}
            displayClass={null}
            itemKey={1}
            key="1"
            label="EDUC"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="EDUC"
          />
          <FilterOption
            active={false}
            count={8}
            displayClass={null}
            itemKey={2}
            key="2"
            label="HEAL"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="HEAL"
          />
          <FilterOption
            active={false}
            count={7}
            displayClass={null}
            itemKey={3}
            key="3"
            label="Ukjent"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="Ukjent"
          />
          <FilterOption
            active={false}
            count={1}
            displayClass={null}
            itemKey={4}
            key="4"
            label="AGRI"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="AGRI"
          />
          <div>
            <Collapse
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              transitionAppear={false}
              unmountOnExit={false}
            >
              <div />
            </Collapse>
            <button
              className="fdk-toggleList"
              onClick={[Function]}
            >
              Vis flere
            </button>
          </div>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`should render FilterBox correctly with activeFilter 1`] = `
<div
  className="fdk-panel--filter"
>
  <div
    className="fdk-panel__header"
  >
    <button
      className="fdk-toggleFilter fdk-p-0"
      onClick={[Function]}
    >
      <i
        className="fa mr-2 fa-angle-up"
      />
      <span>
        Testbox
      </span>
    </button>
  </div>
  <Collapse
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    transitionAppear={false}
    unmountOnExit={false}
  >
    <div
      className="fdk-panel__content"
    >
      <div
        className="fdk-items-list"
      >
        <div>
          <FilterOption
            active={true}
            count={104}
            displayClass={null}
            itemKey={0}
            key="0"
            label="GOVE"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="GOVE"
          />
          <FilterOption
            active={false}
            count={11}
            displayClass={null}
            itemKey={1}
            key="1"
            label="EDUC"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="EDUC"
          />
          <FilterOption
            active={false}
            count={8}
            displayClass={null}
            itemKey={2}
            key="2"
            label="HEAL"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="HEAL"
          />
          <FilterOption
            active={false}
            count={7}
            displayClass={null}
            itemKey={3}
            key="3"
            label="Ukjent"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="Ukjent"
          />
          <FilterOption
            active={false}
            count={1}
            displayClass={null}
            itemKey={4}
            key="4"
            label="AGRI"
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="AGRI"
          />
          <div>
            <Collapse
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              transitionAppear={false}
              unmountOnExit={false}
            >
              <div />
            </Collapse>
            <button
              className="fdk-toggleList"
              onClick={[Function]}
            >
              Vis flere
            </button>
          </div>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`should render FilterBox correctly with activeFilter 2`] = `
<div
  className="fdk-panel--filter"
>
  <div
    className="fdk-panel__header"
  >
    <button
      className="fdk-toggleFilter fdk-p-0"
      onClick={[Function]}
    >
      <i
        className="fa mr-2 fa-angle-up"
      />
      <span>
        Testbox
      </span>
    </button>
  </div>
  <Collapse
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    transitionAppear={false}
    unmountOnExit={false}
  >
    <div
      className="fdk-panel__content"
    >
      <div
        className="fdk-items-list"
      />
    </div>
  </Collapse>
</div>
`;
